pydis_site/apps/content/resources/guides/pydis-guides/contributing/bot.md

```diff
@@ -37,21 +36,113 @@ If your bot fails to start with a `PrivilegedIntentsRequired` exception, this in
+
+**Note**: The `.env` file will be ignored by commits.
+##### 1.2 Setting up the script environment
+The bootstrapping script is Python code like any other. To run it locally, you will need the right version of Python with the necessary packages installed:
 
-#### config.yml
-Entering the directory of the cloned code, you will find a file named `config-default.yml`.
+1. Make sure you follow steps `1` and `2` [here](#setting-up-a-development-environment)
+2. [Install the `config-bootstrap` dependency group](../installing-project-dependencies#installing-specific-dependency-groups).
+
+#### 2. Running the script
+
+Once the script setup phase is complete, all that is left is to run it.
+To do this, you'll simply need to run the `configure` poetry task like this
+
+```shell
+$ poetry run task configure
+```
+
+or, without poetry and from the root directory
+
+```shell
+python3 -m bootstrap_config
+```
+
+Once the script has finished running, you'll notice the creation of a new file called [`.env.server`](#envserver) at your project's root directory.
+This file will contain the extracted ids from your newly created server which are necessary for your bot to run.
+
+**Congratulations**, you have finished the configuration & can now start [running your bot](#run-it)
+
+#### Manual configuration
+##### .env.server
+All server configuration values are saved in a file called `.env.server`, which needs to be at the root directory of the cloned code.
 This file contains the various configurations we use to make the bot run on the Python Discord server, such as channel and role IDs, and the emojis it works with.
 It also contains configurations such as how long it takes for a help channel to time out, and how many messages a user needs to voice-verify.
 
-To run the bot in your test server, you will need to override some of those configurations.
-Create and open a new file in the directory called `config.yml`. Alternatively, copy the `config-default.yml` file and rename the copy to `config.yml`.
-The bot will first look at the items in `config.yml`, and will fall back to `config-default.yml` only if necessary. Note that you don't have to specify all items in `config.yml`, just the ones you want to override such as channel IDs.
+If you decided to use the bootstrapping script, you'll find that this file has already been created (which we recommend),
+otherwise you'll need to create it manually.
+
+To run the bot in your test server, you will **only** need to add the **necessary** configuration values for the channels/roles/categories, etc.
+that you'll be using for testing
+
+Let's take an example where we suppose we'll only be testing a feature that needs the `announcements` channel.
+
+`constants.py`
+
+```py
+
+class EnvConfig:
+    # Defines from where & how Pydantic will be looking for env variables
+    ...
+
+class _Channels(EnvConfig):
+
+    EnvConfig.Config.env_prefix = "channels."
+
+    announcements = 123
+    changelog = 456
```

Comment:
Are these mock channel IDs? If so, I think they should be 19 digits, so they look like real IDs.

Reply:
C'est bon ! 1c23a1715496332a25f2548e208e4e09c31d8be9